llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -175,15 +178,17 @@ static uint64_t getMinimalExtentFrom(const Value &V,
   // If queried with a precise location size, we assume that location size to be
   // accessed, thus valid.
   if (LocSize.isPrecise())
-    DerefBytes = std::max(DerefBytes, LocSize.getValue());
-  return DerefBytes;
+    DerefBytes = std::max(DerefBytes, LocSize.getValue().getKnownMinValue());
+  // FIXME: Is this right?
```

Comment:
I don't think it is. You might combine the result from getPointerDereferenceableBytes with the scalable from from LocSize. I think you should just drop the scalable flag from the result here (this is conservatively correct).

Reply:
ok